.pylintrc

```diff
@@ -21,12 +21,16 @@ disable =
     bad-continuation,
     invalid-name,
     line-too-long,
-    # Algorithms and neural networks generally have a lot of variables
+    # Algorithms and neural networks generally have complex logic
     too-many-instance-attributes,
     too-many-arguments,
     too-many-locals,
+    too-many-branches,
+    too-many-statements,
     # Detection seems buggy or unhelpful
     duplicate-code,
+    # Helper functions without a self use are common in base class
+    no-self-use
```

Comment:
i'm not convinced about this one. Perhaps this is a sign they should be moved into the module as helper functions. W

Reply:
I will do so.

`no-self-use` is added mainly because `BaseAlgo.train()` calls `process_samples()` and some children algorithm may want to overwrite it. If `process_samples()` is moved out of `BaseAlgo` class, overwritting is not possible.